File: apps/wearos/app/src/main/java/com/togplay/vaimarcia/wear/playback/PlaybackEngine.kt
package com.togplay.vaimarcia.wear.playback

/** Contrato comum implementado tanto pelo engine DIRECT quanto pelo RELAY, para que a
 *  camada de UI nunca precise ramificar com base na estratégia — ela apenas pede ao engine
 *  selecionado pelo resolver para reproduzir. */
interface PlaybackEngine {
    suspend fun play(audioId: String): PlaybackResult
}

sealed class PlaybackResult {
    data object Started : PlaybackResult()
    data class Failed(val reason: String) : PlaybackResult()
}

File: apps/watchos/VaiMarciaWatch Watch App/Playback/PlaybackEngine.swift
import Foundation

/// Contrato comum implementado tanto pelo engine DIRECT quanto pelo RELAY, para que a
/// camada de UI nunca precise ramificar com base na estratégia — ela apenas pede ao engine
/// selecionado pelo resolver para reproduzir.
protocol PlaybackEngine {
    func play(audioId: String, completion: @escaping (PlaybackResult) -> Void)
}

enum PlaybackResult {
    case started
    case failed(reason: String)
}
